libs/core/src/lib/flexible-column-layout/flexible-column-layout.component.ts

```diff
@@ -0,0 +1,421 @@
+  styleUrls: ['./flexible-column-layout.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class FlexibleColumnLayoutComponent implements AfterViewInit, OnChanges, OnDestroy, OnInit {
+
+  /**
+   * the template that provides the content of the first/start/left column
+   */
+  @ContentChild('startColumn')
+  startColumn: TemplateRef<any>;
+
+  /**
+   * the template that provides the content of the middle column
+   */
+  @ContentChild('midColumn')
+  midColumn: TemplateRef<any>;
+
+  /**
+   * the template that provides the content of the last/end/right column
+   */
+  @ContentChild('endColumn')
+  endColumn: TemplateRef<any>;
+  
+  /**
+   * the background design of the component
+   * Options include: 'solid' | 'translucent' | 'transparent'
+   * The default is set to 'solid'
+   */
+  @Input()
+  backgroundDesign: 'solid' | 'translucent' | 'transparent' = 'solid';
+
+  /**
+   * the layout of the component
+   * Options include: 'OneColumnStartFullScreen' | 'OneColumnMidFullScreen' |
+   * 'OneColumnEndFullScreen' | 'TwoColumnsStartExpanded' | 'TwoColumnsMidExpanded' |
+   * 'TwoColumnsEndExpanded' | 'ThreeColumnsMidExpanded' | 'ThreeColumnsEndExpanded' |
+   * 'ThreeColumnsStartMinimized' | 'ThreeColumnsEndMinimized'
+   */
+  @Input()
+  layout: Layout = ONE_COLUMN_START_FULL_SCREEN;
+
+  /**
+   * the event emitted on layout value cchange
+   */
+  @Output()
+  layoutChange: EventEmitter<Layout> = new EventEmitter<Layout>();
+
+  /**
+   * user defined break point for MD screens
+   * the default (Fiori 3) value is 599
+   */
+  @Input()
+  mdBreakPoint = 599;
+
+  /**
+   * user defined break point for LG screens
+   * the default (Fiori 3) value is 1023
+   */
+  @Input()
+  lgBreakPoint = 1023;
+
+  /**
+   * user defined break point for XL screens
+   * the default (Fiori 3) value is 1439
+   */
+  @Input()
+  xlBreakPoint = 1439;
+
+  /**
+   * user defined onResize function
+   */
+  @Input()
+  customOnResizeFunction: () => void;
+
+  /**
+   * left column separator value (between start and middle columns)
+   * that specifies the direction of the arrow and
+   * if the separator is visible
+   * Options include: 'left', 'right' and null
+   */
+  _leftColumnSeparator: ColumnSeparatorValue = null;
+
+  /**
+   * right column separator value (between middle and end columns)
+   * that speccifies the direction of the arrow and
+   * if the separator is visible
+   * Options include: 'left', 'right' and null
+   */
+  _rightColumnSeparator: ColumnSeparatorValue = null;
+
+  /** 
+   * @hidden 
+   * allows to keep track of the previos layout 
+   * so we can go back to it on window resize
+   */
+  private _previousLayout: Layout = this.layout;
+
+  /** @hidden */
+  private _screenSize: ScreenSize = LG_SCREEN_SIZE;
+
+  /** @hidden */
+  private _subscriptions = new Subscription();
+
+  /** 
+   * @hidden 
+   * mappting of the layout name and the column layout in %
+   */
+  private _layouts = {
```

Comment:
With `FlexibleColumnSettings` interface you can add type there: `_layouts: {[key: string]: FlexibleColumnSettings} = {`